api/test/unit/views/media_views_test.py

```diff
@@ -241,3 +243,27 @@ def test_thumb_full_size(api_client, media_type, media_factory, requests):
         assert (
             entry in requests.sent_requests[1].request.url
         ), f"{entry} not found in prepared request url: {requests.sent_requests[0].request.url}"
+
+
+@pytest.fixture
+def redis(monkeypatch) -> FakeRedis:
+    fake_redis = FakeRedis()
+    monkeypatch.setattr("catalog.api.views.media_views.cache", fake_redis)
+
+    yield fake_redis
+
+    fake_redis.client().close()
+
+
+@pytest.mark.parametrize("count", [1, 3])
+def test_thumb_timeout(redis, count: int):
+    with patch("requests.get", side_effect=ReadTimeout()):
+        for idx in range(count):
+            with pytest.raises(UpstreamThumbnailException):
+                MediaViewSet._thumbnail_proxy_comm(
+                    "test", params={"url": f"https://example.com/image/{idx}"}
+                )
+
+    # Conversion to ``int`` required because of bug:
+    # https://github.com/cunla/fakeredis-py/issues/58
```

Comment:
Woah! It looks like this is closed upstream now, do we still need to do that conversion or does it need to be brought to a different repo?

Reply:
This was closed as "won't fix" (intended behaviour) so we need to do this conversion.